src/main/java/com/ops/ops/modules/team/application/TeamCommandService.java

```diff
@@ -75,7 +80,11 @@ public void savePreviewImages(Long teamId, List<MultipartFile> images) {
     }
 
     public void deleteTeam(final Long teamId) {
-        teamRepository.delete(teamConvenience.getValidateExistTeam(teamId));
+        final Team team = teamConvenience.getValidateExistTeam(teamId);
+        Long leaderMemberId = teamQueryService.getLeaderIdByTeamId(teamId);
+        memberConvenience.changeLeaderRoleToRegularMember(leaderMemberId);
+        teamMemberConvenience.deleteAllTeamMembersByTeamId(teamId);
+        teamRepository.delete(team);
```

Comment:
```suggestion
    public void deleteTeam(final Long teamId) {
        final Team team = teamConvenience.getValidateExistTeam(teamId);
        final Member leader = memberConvenience.findByName(team.getLeaderName());
        leader.updateRole();
        teamRepository.delete(team);
    }
```

다른 수정 없이 leaderName으로 리더 찾고, leader의 Role만 바꿔주면 될 것 같아요.
updateRoled은 Member Entity에 적으시면 될 것 같습니다!

Reply:
엇 그런데 member의 이름이 중복 가능한데 Team의 teamMembers에서 리더를 찾아야 하지 않을까요??

Reply:
아 대회가 늘어나서 그럴 수도 있겠네요.  그렇다면 팀 아이디로 리더 아이디 찾아서(`getLeaderIdByTeamId`를 `teamConvenience`로 이동) -> 리더 아이디로 멤버를 찾아서 (`memberConvenience`에 `findById`) -> 권한 변경(leader.updateRole) 하는 건 어떨까요??

Reply:
넵 그렇게 수정할게요! 감사합니다